File: android-app/app/src/main/java/com/plantgotchi/app/analytics/Analytics.kt
// android-app/app/src/main/java/com/plantgotchi/app/analytics/Analytics.kt
package com.plantgotchi.app.analytics

import com.posthog.PostHog

enum class LogLevel { INFO, WARN, ERROR }

object Analytics {
    fun track(event: String, properties: Map<String, Any> = emptyMap()) {
        try { PostHog.capture(event, properties = properties) } catch (_: Exception) {}
    }

    fun identify(userId: String, traits: Map<String, Any>) {
        try { PostHog.identify(userId, userProperties = traits) } catch (_: Exception) {}
    }

    fun reset() {
        try { PostHog.reset() } catch (_: Exception) {}
    }

    fun captureException(throwable: Throwable, context: Map<String, Any> = emptyMap()) {
        try {
            val props = mutableMapOf<String, Any>(
                "\$exception_type" to (throwable::class.simpleName ?: "Unknown"),
                "\$exception_message" to (throwable.message ?: ""),
                "\$exception_stack_trace_raw" to throwable.stackTraceToString(),
            )
            props.putAll(context)
            PostHog.capture("\$exception", properties = props)
        } catch (_: Exception) {}
    }

    fun log(level: LogLevel, message: String, context: Map<String, Any> = emptyMap()) {
        try {
            val props = mutableMapOf<String, Any>(
                "level" to level.name.lowercase(),
                "message" to message,
            )
            props.putAll(context)
            PostHog.capture("log_entry", properties = props)
        } catch (_: Exception) {}
    }
}

File: website-astro/src/lib/analytics.ts
import posthog from 'posthog-js';

export const Analytics = {
  track(event: string, properties?: Record<string, any>): void {
    posthog.capture(event, properties);
  },

  identify(userId: string, traits: Record<string, any>): void {
    posthog.identify(userId, traits);
  },

  reset(): void {
    posthog.reset();
  },

  captureException(error: Error, context?: Record<string, any>): void {
    posthog.capture('$exception', {
      $exception_type: error.constructor.name,
      $exception_message: error.message,
      $exception_stack_trace_raw: error.stack ?? '',
      ...context,
    });
  },

  log(level: 'info' | 'warn' | 'error', message: string, context?: Record<string, any>): void {
    posthog.capture('log_entry', {
      level,
      message,
      ...context,
    });
  },
};
